### ISSUE TITLE
Segfault when starting ert

### ISSUE THREAD

**Issue by frode-aarstad** [equinor/ert#3363]
**To Reproduce**
1. `pip install ert`
2. `export LANG=en_US.ISO8859-15 ; ert gui test-data/local/snake-oil/snake-oil.ert`

**Enviromment**
 - ERT - main
 - Python version 3.6/3.8
 - RHEL

**Additional context**
After some initial testing it seems like the problem might be in the `pydantic 1.9.0` library when having a "non standard" language setting. `pydantic 1.8.2` seems to work fine.

The following code gives some more info:

```
export LANG=en_US.ISO8859-15 ; python -c "import faulthandler; faulthandler.enable(); from open_petro_elastic.__main__ import make_input"
Fatal Python error: Segmentation fault

Current thread 0x00007f620789c740 (most recent call first):
  File "/prog/res/komodo/2022.04.00-py36-rhel7/root/lib/python3.6/site-packages/open_petro_elastic/config/constituent.py", line 37 in Constituent
  File "/prog/res/komodo/2022.04.00-py36-rhel7/root/lib/python3.6/site-packages/open_petro_elastic/config/constituent.py", line 15 in <module>
  File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 678 in exec_module
  File "<frozen importlib._bootstrap>", line 665 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 955 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 971 in _find_and_load
  File "/prog/res/komodo/2022.04.00-py36-rhel7/root/lib/python3.6/site-packages/open_petro_elastic/config/__init__.py", line 14 in <module>
  File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 678 in exec_module
  File "<frozen importlib._bootstrap>", line 665 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 955 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 971 in _find_and_load
  File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 941 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 971 in _find_and_load
  File "/prog/res/komodo/2022.04.00-py36-rhel7/root/lib/python3.6/site-packages/open_petro_elastic/__main__.py", line 16 in <module>
  File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 678 in exec_module
  File "<frozen importlib._bootstrap>", line 665 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 955 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 971 in _find_and_load
  File "<string>", line 1 in <module>



```

**Comment by BjarneHerland** [comment]
This looks like a problem with pydantic on RHEL.

Even without enabling komodo on RGS, setting LANG like described, [one of their official examples](https://pydantic-docs.helpmanual.io/usage/validators/#validate-always) fails. With Python3.6 it fails when parsing the validator, but with Python3.8 it fails when applying the validator in the second print() (add `import faulthandler;faulthandler.enable()` as first line in example to see linenumbers).

If not modifying LANG (leaving it to `en_US.UTF-8`) this example works fine with both 3.6 and 3.8. Moreover, it works fine my Mac with both LANG-settings.

As mentioned, downgrading to pydantic 1.8.2 works fine.

**Comment by oysteoh** [comment]
@xjules this i guess also relates to your problems on your mac? Could you downgrade pydantic to 1.8.2 and check?

**Comment by xjules** [comment]
No it didn't help the mac issue (at least in my case):
```sh
(/Users/JPARU/.local/py38-gui) ert:~ $ export LANG="en_US.UTF-8"
(/Users/JPARU/.local/py38-gui) ert:~ $ python -c "from qtpy.QtCore import QLocale; print(QLocale().decimalPoint())"
,
(/Users/JPARU/.local/py38-gui) ert:~ $ pip show pydantic
Name: pydantic
Version: 1.8.2
...
(/Users/JPARU/.local/py38-gui) ert:~ $ locale
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="C"
LC_TIME="en_US.UTF-8"
LC_ALL=
```

**Comment by BjarneHerland** [comment]
> This looks like a problem with pydantic on RHEL.
> 
> Even without enabling komodo on RGS, setting LANG like described, [one of their official examples](https://pydantic-docs.helpmanual.io/usage/validators/#validate-always) fails. With Python3.6 it fails when parsing the validator, but with Python3.8 it fails when applying the validator in the second print() (add `import faulthandler;faulthandler.enable()` as first line in example to see linenumbers).
> 
> If not modifying LANG (leaving it to `en_US.UTF-8`) this example works fine with both 3.6 and 3.8. Moreover, it works fine my Mac with both LANG-settings.
> 
> As mentioned, downgrading to pydantic 1.8.2 works fine.

@sondreso any preference for how to proceed with this?

**Comment by sondreso** [comment]
Seems like a very unfortunatebug :thinking:  I think we should create a minimal reproducing example, and report it to Pydantic.

**Comment by BjarneHerland** [comment]
I can do that, but someone should decide whether to downgrade pydantic-version in komodo or lnot. Not sure how extensive this problem among users.

**Comment by BjarneHerland** [comment]
Perhaps something along the lines of #3296 is a way to handle this?

**Comment by oysteoh** [comment]
Then i'll close this issue as fixed - and the long term solution is #3296
